packages/web3-eth/src/errors.ts

```diff
@@ -293,6 +294,24 @@ export class TransactionPollingTimeoutError extends Web3Error {
 	}
 }
 
+export class TransactionBlockTimeoutError extends Web3Error {
+	public code = ERR_TX_BLOCK_TIMEOUT;
+
+	public constructor(value: {
+		starterBlockNumber: number;
+		numberOfBlocks: number;
+		transactionHash?: Bytes;
+	}) {
+		super(
+			`Transaction started at ${value.starterBlockNumber} but was not mined within ${
+				value.numberOfBlocks
+			} blocks, please make sure your transaction was properly sent and there no pervious pending transaction for the same account. However, be aware that it might still be mined!\n\tTransaction Hash: ${
```

Comment:
```suggestion
			} blocks, please make sure your transaction was properly sent and there is no previous pending transaction for the same account. However, be aware that it might still be mined!\n\tTransaction Hash: ${
```